src/gurt/fault_inject.c

```diff
@@ -82,63 +79,117 @@ fa_op_rec_decref(struct d_hash_table *htab, d_list_t *rlink)
 }
 
 static d_hash_table_ops_t fa_table_ops = {
-	.hop_key_cmp	= fa_op_key_cmp,
-	.hop_rec_hash	= fa_op_rec_hash,
-	.hop_rec_decref	= fa_op_rec_decref,
-	.hop_rec_free	= fa_op_rec_free,
+    .hop_key_cmp    = fa_op_key_cmp,
+    .hop_rec_hash   = fa_op_rec_hash,
+    .hop_rec_decref = fa_op_rec_decref,
+    .hop_rec_free   = fa_op_rec_free,
 };
 
 struct d_fi_gdata_t {
-	unsigned int		  dfg_refcount;
-	unsigned int		  dfg_inited;
-	pthread_rwlock_t	  dfg_rwlock;
-	struct d_hash_table	  dfg_fa_table;
+	unsigned int        dfg_refcount;
+	unsigned int        dfg_inited;
+	pthread_rwlock_t    dfg_rwlock;
+	struct d_hash_table dfg_fa_table;
+	bool                dfg_thread_default;
 };
 
-/**
- * global switch for fault injection. zero globally turns off fault injection,
- * non-zero turns on fault injection
+/* global switch for fault injection. zero globally turns off fault injection, non-zero turns on
+ * fault injection
+ */
+static uint32_t            d_fault_inject_seed;
+static struct d_fi_gdata_t d_fi_gdata;
+static pthread_once_t      d_fi_gdata_init_once = PTHREAD_ONCE_INIT;
+
+/* Check for and set thread specific enabled  flag.  This is a single function that returns two
+ * values so it can access a single _thread local.
+ *
+ * 0 - no value set;
+ * 1 - disabled;
+ * >2 - enabled;
```

Comment:
would be more readable to add defines for those, e.g. FAULTS_NOT_SET, FAULTS_DISABLED, FAULTS_ENABLED or some such.

Reply:
If this wasn't a single static function in a header file I'd agree.